railties/test/application/rake/dbs_test.rb

```diff
@@ -249,15 +249,21 @@ def with_bad_permissions
         end
 
         test "db:drop failure because bad permissions" do
+          ENV["DISABLE_DATABASE_ENVIRONMENT_CHECK"] = "1"
```

Comment:
There's a `with_env` helper.

Reply:
Nice find, fixed in [7f22183](https://github.com/rails/rails/pull/49930/commits/7f221837db0dc7a8d5219ef72d64f82910422f33).